File: tests/0104_maximum_depth_of_binary_tree_test.py
import importlib
import pytest
from algorithms.tree_node import make_tree

module = importlib.import_module("algorithms.0104_maximum_depth_of_binary_tree")


def test_max_depth():
    t = make_tree([3, 9, 20, None, None, 15, 7])

    s = module.Solution()
    assert s.maxDepth(t[0]) == 3

File: tests/0104_maximum_depth_of_binary_tree_test.cpp
#include "0104_maximum_depth_of_binary_tree.cpp"
#include <doctest.h>

TEST_CASE("maxDepth")
{
    auto t = tree::make_tree({3, 9, 20, std::nullopt, std::nullopt, 15, 7});

    Solution s = Solution();
    REQUIRE(s.maxDepth(t[0].get()) == 3);
}
